torch/fx/passes/reinplace.py

```diff
@@ -0,0 +1,303 @@
+import torch
+import torch.fx as fx
+from torch.fx._compatibility import compatibility
+
+from collections import defaultdict
+from typing import Iterable
+import itertools
+
+__all__ = ['reinplace']
```

Comment:
This is the PassBase interface we wish to standardize... Appreciate any feedback 

https://github.com/pytorch/pytorch/blob/729c56d1363a61ec5324743a6a3a3098a084153e/torch/fx/passes/infra/pass_base.py

Reply:
I guess postcondition would be a good place to add the "check that the fake tensor annotations are still up to date after the transform"